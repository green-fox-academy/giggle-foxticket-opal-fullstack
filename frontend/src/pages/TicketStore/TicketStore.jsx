import React from 'react';
import TicketList from '../../components/TicketTypes/TicketList';

const TicketStore = () => {
  return (
    <div className="ticket-store">
      <TicketList />
    </div>
  );
};

export default TicketStore;
