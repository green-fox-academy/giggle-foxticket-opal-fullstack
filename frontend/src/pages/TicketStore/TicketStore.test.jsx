import React from 'react';
import TicketStore from './TicketStore';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';


describe('Connect component to Redux store', () => {
  const mockStore = configureStore([]);
  const user = { name: 'Someone' }
  const store = mockStore({ user });

  it('should render with given state from Redux store', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <TicketStore />
        </Provider>
      </BrowserRouter>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
