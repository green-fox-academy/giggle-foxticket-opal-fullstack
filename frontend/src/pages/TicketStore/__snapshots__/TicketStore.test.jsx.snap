// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ticket-store"
>
  <div
    className="ticket-list-container"
  >
    <div
      className="ticket-list"
    >
      <h1
        className="main-title"
      >
        Ticket Types
      </h1>
      <div
        className="admin_container"
      >
        <div
          className="admin_icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 96h-48V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56v400c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24v-42.11l80.606-35.977C429.396 365.063 448 336.388 448 304.86V176c0-44.112-35.888-80-80-80zm16 208.86a16.018 16.018 0 0 1-9.479 14.611L320 343.805V160h48c8.822 0 16 7.178 16 16v128.86zM208 384c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16zm-96 0c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16z"
            />
          </svg>
        </div>
        <div
          className="admin_content"
        >
          <h1
            className="admin_title"
          >
            Ticket-Types A
          </h1>
          <p
            className="admin_description"
          >
            Lorem Ipsum has been the industry s standard dummy
          </p>
        </div>
        <div
          className="btns"
        >
          <button
            className="btn btn--danger--solid btn--medium"
          >
            BUY
          </button>
        </div>
      </div>
      <div
        className="admin_container"
      >
        <div
          className="admin_icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm117.8-146.4c-10.2-8.5-25.3-7.1-33.8 3.1-20.8 25-51.5 39.4-84 39.4s-63.2-14.3-84-39.4c-8.5-10.2-23.7-11.5-33.8-3.1-10.2 8.5-11.5 23.6-3.1 33.8 30 36 74.1 56.6 120.9 56.6s90.9-20.6 120.9-56.6c8.5-10.2 7.1-25.3-3.1-33.8zM168 240c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160-60c-25.7 0-55.9 16.9-59.9 42.1-1.7 11.2 11.5 18.2 19.8 10.8l9.5-8.5c14.8-13.2 46.2-13.2 61 0l9.5 8.5c8.5 7.4 21.6.3 19.8-10.8-3.8-25.2-34-42.1-59.7-42.1z"
            />
          </svg>
        </div>
        <div
          className="admin_content"
        >
          <h1
            className="admin_title"
          >
            Ticket-Types B
          </h1>
          <p
            className="admin_description"
          >
            Since the 1500s,when an unknown printer took a galley
          </p>
        </div>
        <div
          className="btns"
        >
          <button
            className="btn btn--danger--solid btn--medium"
          >
            BUY
          </button>
        </div>
      </div>
      <div
        className="admin_container"
      >
        <div
          className="admin_icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 0H112c-4 0-7.8 2-10 5.4L2 152.6c-2.9 4.4-2.6 10.2.7 14.2l276 340.8c4.8 5.9 13.8 5.9 18.6 0l276-340.8c3.3-4.1 3.6-9.8.7-14.2L474.1 5.4C471.8 2 468.1 0 464 0zm-19.3 48l63.3 96h-68.4l-51.7-96h56.8zm-202.1 0h90.7l51.7 96H191l51.6-96zm-111.3 0h56.8l-51.7 96H68l63.3-96zm-43 144h51.4L208 352 88.3 192zm102.9 0h193.6L288 435.3 191.2 192zM368 352l68.2-160h51.4L368 352z"
            />
          </svg>
        </div>
        <div
          className="admin_content"
        >
          <h1
            className="admin_title"
          >
            Ticket-Types C
          </h1>
          <p
            className="admin_description"
          >
            Remaining  essentially unchanged. It was popularised
          </p>
        </div>
        <div
          className="btns"
        >
          <button
            className="btn btn--danger--solid btn--medium"
          >
            BUY
          </button>
        </div>
      </div>
      <div
        className="admin_container"
      >
        <div
          className="admin_icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M575.7 280.8C547.1 144.5 437.3 62.6 320 49.9V32c0-17.7-14.3-32-32-32s-32 14.3-32 32v17.9C138.3 62.6 29.5 144.5.3 280.8c-2.2 10.1 8.5 21.3 18.7 11.4 52-55 107.7-52.4 158.6 37 5.3 9.5 14.9 8.6 19.7 0 20.2-35.4 44.9-73.2 90.7-73.2 58.5 0 88.2 68.8 90.7 73.2 4.8 8.6 14.4 9.5 19.7 0 51-89.5 107.1-91.4 158.6-37 10.3 10 20.9-1.3 18.7-11.4zM256 301.7V432c0 8.8-7.2 16-16 16-7.8 0-13.2-5.3-15.1-10.7-5.9-16.7-24.1-25.4-40.8-19.5-16.7 5.9-25.4 24.2-19.5 40.8 11.2 31.9 41.6 53.3 75.4 53.3 44.1 0 80-35.9 80-80V301.6c-9.1-7.9-19.8-13.6-32-13.6-12.3.1-22.4 4.8-32 13.7z"
            />
          </svg>
        </div>
        <div
          className="admin_content"
        >
          <h1
            className="admin_title"
          >
            Ticket-Types D
          </h1>
          <p
            className="admin_description"
          >
            Contrary to popular Lorem Ipsum is not  random text
          </p>
        </div>
        <div
          className="btns"
        >
          <button
            className="btn btn--danger--solid btn--medium"
          >
            BUY
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-list-container"
  >
    <div
      className="ticket-list"
    >
      <h1
        className="main-title"
      >
        My tickets
      </h1>
      <div
        className="admin_container"
      >
        <div
          className="admin_icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 96h-48V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56v400c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24v-42.11l80.606-35.977C429.396 365.063 448 336.388 448 304.86V176c0-44.112-35.888-80-80-80zm16 208.86a16.018 16.018 0 0 1-9.479 14.611L320 343.805V160h48c8.822 0 16 7.178 16 16v128.86zM208 384c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16zm-96 0c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16z"
            />
          </svg>
        </div>
        <div
          className="admin_content"
        >
          <h1
            className="admin_title"
          >
            Single ticket
          </h1>
          <p
            className="admin_description"
          >
            Not validated
          </p>
        </div>
        <div
          className="btns"
        >
          <button
            className="btn btn--danger--solid btn--medium"
          >
            SHOW
          </button>
        </div>
      </div>
      <div
        className="admin_container"
      >
        <div
          className="admin_icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm117.8-146.4c-10.2-8.5-25.3-7.1-33.8 3.1-20.8 25-51.5 39.4-84 39.4s-63.2-14.3-84-39.4c-8.5-10.2-23.7-11.5-33.8-3.1-10.2 8.5-11.5 23.6-3.1 33.8 30 36 74.1 56.6 120.9 56.6s90.9-20.6 120.9-56.6c8.5-10.2 7.1-25.3-3.1-33.8zM168 240c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160-60c-25.7 0-55.9 16.9-59.9 42.1-1.7 11.2 11.5 18.2 19.8 10.8l9.5-8.5c14.8-13.2 46.2-13.2 61 0l9.5 8.5c8.5 7.4 21.6.3 19.8-10.8-3.8-25.2-34-42.1-59.7-42.1z"
            />
          </svg>
        </div>
        <div
          className="admin_content"
        >
          <h1
            className="admin_title"
          >
            Single ticket
          </h1>
          <p
            className="admin_description"
          >
            Valid until ...
          </p>
        </div>
        <div
          className="btns"
        >
          <button
            className="btn btn--danger--solid btn--medium"
          >
            SHOW
          </button>
        </div>
      </div>
      <div
        className="admin_container"
      >
        <div
          className="admin_icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 0H112c-4 0-7.8 2-10 5.4L2 152.6c-2.9 4.4-2.6 10.2.7 14.2l276 340.8c4.8 5.9 13.8 5.9 18.6 0l276-340.8c3.3-4.1 3.6-9.8.7-14.2L474.1 5.4C471.8 2 468.1 0 464 0zm-19.3 48l63.3 96h-68.4l-51.7-96h56.8zm-202.1 0h90.7l51.7 96H191l51.6-96zm-111.3 0h56.8l-51.7 96H68l63.3-96zm-43 144h51.4L208 352 88.3 192zm102.9 0h193.6L288 435.3 191.2 192zM368 352l68.2-160h51.4L368 352z"
            />
          </svg>
        </div>
        <div
          className="admin_content"
        >
          <h1
            className="admin_title"
          >
            Single ticket
          </h1>
          <p
            className="admin_description"
          >
            Expired
          </p>
        </div>
        <div
          className="btns"
        >
          <button
            className="btn btn--danger--solid btn--medium"
          >
            SHOW
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
